src/serde/tree_cache.rs

```diff
@@ -0,0 +1,744 @@
+use super::bytes32::{hash_blobs, Bytes32};
+use super::{ChildPos, PathBuilder};
+use crate::allocator::{Allocator, NodePtr, SExp};
+use crate::serde::serialized_length_atom;
+use crate::serde::RandomState;
+use crate::serde::VisitedNodes;
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+
+const MIN_SERIALIZED_LENGTH: u64 = 4;
+
+#[derive(Clone, Debug)]
+struct NodeEntry {
+    /// the tree hash of this node. It may be None if it or any of its children
```

Comment:
s/children/descendents/

Reply:
```suggestion
    /// the tree hash of this node. It may be None if it or any of its children
    /// the tree hash of this node. It may be None if it or any of its descendants
```